pyvista/core/composite.py

```diff
@@ -39,7 +39,7 @@
 
     from ._typing_core import NumpyArray
 
-_TypeMultiBlockLeaf = Union['MultiBlock', DataSet, None]
+_TypeMultiBlockLeaf = Union['MultiBlock', DataSet]
```

Comment:
This change is a bit concerning since `None` was only recently added here. https://github.com/pyvista/pyvista/pull/6852
`None` is a valid leaf, so I think we should always expect to handle `None` blocks (and have `mypy` warn us when we're not handling it properly)